### PR TITLE
Remove deprecated Win32API

### PR DESCRIPTION
**Pull request by jaymalasinha**
Signed-off-by: Jaymala Sinha <jsinha@chef.io>

### Description

Remove the deprecated Win32API as it is now included in chef.
Update to chef client version >=14

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: libraries/version.rb [hunk 1]
```diff
@@ -186,10 +183,9 @@ def initialize
 
     private
 
-    # Win32API call to GetSystemMetrics(SM_SERVERR2)
     # returns: The build number if the system is Windows Server 2003 R2; otherwise, 0.
     def sm_serverr2
```
Comment (tas50) [inline libraries/version.rb]: We can probably just remove this method since we haven't support 2003 for a long time.

Comment (jaymalasinha) [reply libraries/version.rb]: removed

File: libraries/windows_helper.rb [hunk 2]
```diff
@@ -86,17 +84,6 @@ def cached_file(source, checksum = nil, windows_path = true)
       end
     end
 
-    # Expands the environment variables
-    def expand_env_vars(path)
```
Comment (tas50) [inline libraries/windows_helper.rb]: We probably want this still. I'm sure it's used somewhere.

File: metadata.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@
 license          'Apache-2.0'
 description      'Provides a set of useful Windows-specific primitives.'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version          '5.3.0'
+version          '5.3.1'
```
Comment (tas50) [inline metadata.rb]: Our tooling bumps the versions so this isn't necessary

File: libraries/windows_helper.rb (hunk 2 above)
Comment (jaymalasinha) [reply libraries/windows_helper.rb]: Ref: https://github.com/chef/chef/commit/561c06d451427d444217f6d901a16f30144a6903#diff-6f1a02d1cc16201c682598cb413ebd42L52

**Comment by tas50** [comment]
Thanks @jaymalasinha

**Comment by tas50** [comment]
Released as 5.3.1

**Pull request merged**